src/pipx/venv.py

```diff
@@ -327,8 +342,43 @@ def list_installed_packages(self) -> Set[str]:
         pip_list = json.loads(cmd_run.stdout.strip())
         return set([x["name"] for x in pip_list])
 
+    def _find_entry_point(self, app: str) -> Optional[EntryPoint]:
+        if not self.python_path.exists():
+            return None
+        dists = Distribution.discover(
+            name=self.main_package_name,
+            path=[str(get_site_packages(self.python_path))],
+        )
+        for dist in dists:
+            for ep in dist.entry_points:
+                if ep.group == "pipx.run" and ep.name == app:
```

Comment:
@uranusjr I've just encountered a case where the entry point would not be found due to a path given as app. Can we just drop the requirement that the entry point name must exactly match the application name, and pick any entry point found under the `pipx.run` table?

Reply:
What if there is more than one? Say "a" depends on "b", and they both have entry points? (If there's only one, seems fine, just like we already do for one script)